scanpipe/pipes/d2d.py

```diff
@@ -422,26 +423,28 @@ def create_package_from_purldb_data(project, resources, package_data):
     package_data.pop("uuid", None)
     package_data.pop("dependencies", None)
 
-    lookups = Q()
-    for resource in resources:
-        lookups |= Q(path=resource.path)
-        if resource.is_archive:
-            # This is done to capture the extracted contents of the archive we
-            # matched to. Generally, the archive contents are in a directory
-            # that is the archive path with `-extract` at the end.
-            lookups |= Q(path__startswith=resource.path)
-        elif resource.is_dir:
-            # We add a trailing slash to avoid matching on directories we do not
-            # intend to. For example, if we have matched on the directory with
-            # the path `foo/bar/1`, using the __startswith filter without
-            # including a trailing slash on the path would have us get all
-            # diretories under `foo/bar/` that start with 1, such as
-            # `foo/bar/10001`, `foo/bar/123`, etc., when we just want `foo/bar/1`
-            # and its descendants.
-            path = f"{resource.path}/"
-            lookups |= Q(path__startswith=path)
-
-    resources_qs = project.codebaseresources.to_codebase().filter(lookups)
+    resources_qs = None
+    if not is_empty_queryset(resources):
```

Comment:
Is this check needed at all, the `for resource in resources:` should skip anyway when `resources` is empty, no?

Reply:
`for resources in resources:` does skip when `resources` is empty, but this would cause a bug where we pass an empty `Q()` object into our filter and it returned all CodebaseResources in the `to/` codebase (https://github.com/nexB/scancode.io/blob/main/scanpipe/pipes/d2d.py#L427)

I could skip the `is_empty_queryset` check and just check if `lookups` is not an empty `Q()` before using it.